### PR TITLE
Add a node to convert from DiagnosticArray to ParameterDeclaration

### PR DESCRIPTION
**Pull request by maxime-clem**
## Types of PR

- [X] New Features
- [ ] Upgrade of existing features
- [ ] Bugfix

## Link to the issue

Needed to integrate the planner evaluation metrics (https://tier4.atlassian.net/browse/T4PB-10536) with the scenario runner.

## Description

Add the `diagnostic_converter` package which contains a node to convert `diagnostic_msgs::msg::DiagnosticArray` messages into `openscenario_msgs::msg::ParameterDeclaration`.

Also update the launch file to run the converter and the planning evaluator that publishes `DiagnosticArray` messages.

## How to review this PR.

Checkout branch `10536-planning-evaluator` on the following repositories (in addition to `scenario_simulator_v2`).
- `autoware.iv`: adds the package that publishes planner evaluation metrics as `DiagnosticArray` messages.
- `AutonomousDrivingScenarios`: adds a test scenario which uses a `UserDefinedCondition` based on a metric published by the planning evaluator.

Run command `./run_scenario.sh metric_test.yaml` inside `AutonomousDrivingScenarios/scripts`.
(Make sure you `source ~/autoware/autoware.proj/install/setup.sh` after rebuilding Autoware.iv).

The test scenario `metric_test.yaml` listens to `/diagnostic/planning_evaluator/metrics_stability_mean` and the scenario fails if the corresponding value is greater than `0.2`. The original data is published on topic `/planning_evaluator/metrics_stability_mean` as a `DiagnosticArray` message.

### CODE REVIEW

**Comment by hakuturu583** [comment]
@maxime-clem @yamacir-kit I fixed some small problems. (linters, release notes, merge with master branch etc..)

File: openscenario/diagnostic_converter/README.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Planning Evaluator
+
+## Purpose
+
+This package provides a node to convert `diagnostic_msgs::msg::DiagnosticArray` messages
+into `openscenario_msgs::msg::ParameterDeclaration` messages.
+
+## Inner-workings / Algorithms
+
+The node subscribes to all topics listed in the parameters and assume they publish
```
Comment (kmiya) [inline openscenario/diagnostic_converter/README.md:10]:
```suggestion
The node subscribes to all topics listed in the parameters and assumes they publish
```
Nit: assume -> assumes?

**Comment by maxime-clem** [comment]
@kmiya I wrote the node for ROS `galactic` but in the CI it is built using ROS `foxy`.
Should I rewrite the code to be compatible with `foxy` ?

**Comment by kmiya** [comment]
> @kmiya I wrote the node for ROS `galactic` but in the CI it is built using ROS `foxy`. Should I rewrite the code to be compatible with `foxy` ?

Delegate to: @kenji-miyake

**Comment by kenji-miyake** [comment]
Why me? :thinking: I think it depends on the policy of maintainers of this repository.
But if I would say my personal opinion, supporting both `foxy` and `galactic` is better if it's possible.

**Comment by kmiya** [comment]
Oh, here is scenario_simulator_v2 repository. I confused with Autoware's.
@hakuturu583 would be the person to answer the question.

**Comment by hakuturu583** [comment]
scenario_simulator_v2 currently support galactic and foxy because Autoware.Auto supports foxy.
@maxime-clem Can you modify this code works with galactic and foxy?

**Comment by hakuturu583** [comment]
Sorry for may late reply.

File: test_runner/scenario_test_runner/launch/scenario_test_runner.launch.py [hunk 2]
```diff
@@ -189,6 +189,30 @@ def description():
...
+            ],
+        ),
+        # Node to convert DiagnosticArray messages into ParameterDeclaration messages
+        Node(
+            package="diagnostic_converter",
+            executable="diagnostic_converter",
+            name="diagnostic_converter",
+            output={"stderr": "log", "stdout": "log"},
+            parameters=[{"diagnostic_topics" : ["/diagnostic/planning_evaluator/metrics"]}],
+        ),
```
Comment (maxime-clem) [inline test_runner/scenario_test_runner/launch/scenario_test_runner.launch.py:215]: This code causes the remaining CI errors.
Package `planning_evaluator` is in Autoware.iv which seems unavailable when running the CI.
Should I simply remove this changes to the `scenario_test_runner.launch.py` ?

**Pull request closed without merging**